src/account/sync/mod.rs

```diff
@@ -1620,24 +1614,12 @@ async fn is_dust_allowed(
     let mut dust_outputs_amount: i64 = 0;
 
     // Add outputs from this transaction
-    for output in outputs {
-        match output.1 {
-            // add newly created outputs
-            true => {
-                if output.0 >= DUST_ALLOWANCE_VALUE {
-                    dust_allowance_balance += output.0 as i64;
-                } else {
-                    dust_outputs_amount += 1
-                }
-            }
-            // remove consumed outputs
-            false => {
-                if output.0 >= DUST_ALLOWANCE_VALUE {
-                    dust_allowance_balance -= output.0 as i64;
-                } else {
-                    dust_outputs_amount -= 1;
-                }
-            }
+    for (dust, add_outputs) in outputs {
+        let balance_to_add = if add_outputs { 1 } else { -1 };
```

Comment:
Please rename `balance_to_add`, because it's not the balance
(if we update this part here we should also do it in iota.rs)